objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestParallelProcessing.java

```diff
@@ -0,0 +1,220 @@
+
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeTrue;
+
+public class TestParallelProcessing extends NeuralNetTestBase {
+
+    static ObjectRecognitionCore core;
+    static Vantiq vantiq;
+
+    static final String FAKE_MODEL_DIR = "";
+    static final int CORE_START_TIMEOUT = 10;
+    static final String IP_CAMERA_ADDRESS = "http://207.192.232.2:8000/mjpg/video.mjpg";
+
+    @BeforeClass
+    public static void classSetup() {
+        vantiq = new io.vantiq.client.Vantiq(testVantiqServer);
+        vantiq.setAccessToken(testAuthToken);
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        // Double check that everything was deleted from VANTIQ
+        deleteSource();
+        deleteType();
+        deleteRule();
+    }
+
+    @Test
+    public void testDiscardPolicyDefaultThreadConfig() throws InterruptedException {
+        // The default values for maxRunningThreads and maxQueuedTasks are 10 and 20 respectively
+        // The pollTime is set to 100 (10 fps). Since the TestProcessor sleepTime is set to 5 seconds, we know that 10 fps is impossible
+        // The first 10 tasks should take 5 seconds to complete, as well as the next 10. We know we should have ~20 entries after waiting 10 seconds.
+        // The timing is not exact, so we make sure we are under 30 total, (well below the 100 we would see if the TestProcessor sleepTime was 0).
+        discardPolicyTestHelper(false, 30);
+    }
+
+    @Test
+    public void testDiscardPolicyCustomThreadConfig() throws InterruptedException {
+        // The custom values for maxRunningThreads and maxQueuedTasks are 5 and 10 respectively
+        // The pollTime is set to 100 (10 fps). Since the TestProcessor sleepTime is set to 5 seconds, we know that 10 fps is impossible
+        // The first 5 tasks should take 5 seconds to complete, as well as the next 5. We know we should have ~10 entries after waiting 10 seconds.
+        // The timing is not exact, so we make sure we are under 20 total, (well below the 100 we would see if the TestProcessor sleepTime was 0).
+        discardPolicyTestHelper(true, 20);
+    }
+
+    public void discardPolicyTestHelper(boolean useCustomTaskConfig, int numberOfEntries) throws InterruptedException {
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+
+        // Check that Source, Type, Topic, Procedure and Rule do not already exist in namespace, and skip test if they do
+        assumeFalse(checkSourceExists());
+        assumeFalse(checkTypeExists());
+        assumeFalse(checkRuleExists());
+
+        // Setup a VANTIQ Obj Rec Source, and start running the core
+        setupSource(createSourceDef(useCustomTaskConfig));
+
+        // Create Type to store results
+        setupType();
+
+        // Create Rule to store results in Type
+        setupRule();
+
+        // Wait for 10 seconds
+        Thread.sleep(10000);
```

Comment:
probably add some comment that says something about the source acting independently so we're awaiting it's doing something?  I'm guessing.

Reply:
I'll add a comment in the code. Basically, we wait for 10 seconds while the source polls for images, and then we check the results to make sure that the maxRunningThreads/QueuedTasks work appropriately, and that the DiscardOldestPolicy works as well.